web/src/compose.js

```diff
@@ -253,6 +253,10 @@ export function send_message(request = create_message_object()) {
         drafts.sync_count();
 
         const draft = drafts.draft_model.getDraft(request.draft_id);
+        if (!draft) {
+            return;
+        }
```

Comment:
I think we do unconditionally store this, though, and `drafts.update_draft` is not a network call.

There are two possibilities for what could be going on here:
- Browser local storage is disable. I'm not sure how we handle that, but there's an argument that we should just use an in-memory dictionary as a fallback for that situation rather than returning empty.
- There's some sort of race that allows the user to manipulate/delete the draft object.
Do you understand which it is?

``` js
export function send_message(request = create_message_object()) {                                       
    compose_state.set_recipient_edited_manually(false);                                                 
    compose_state.set_is_content_unedited_restored_draft(false);                                        
    if (request.type === "private") {                                                                   
        request.to = JSON.stringify(request.to);                                                        
    } else {                                                                                            
        request.to = JSON.stringify([request.to]);                                                      
    }                                                                                                   
                                                                                                        
    // Silently save / update a draft to ensure the message is not lost in case send fails.             
    // We delete the draft on successful send.                                                          
    request.draft_id = drafts.update_draft({                                                            
        no_notify: true,                                                                                
        update_count: false,                                                                            
        is_sending_saving: true,                                                                        
    });                                                                                                 
```

Reply:
We don't save draft if the message doesn't have enough content. See `compose_state.has_savable_message_content()`.

Reply:
Probably this is the right fix in that case.